lib/models/user-identity.js

```diff
@@ -121,9 +121,7 @@
                        loopback.getModelByType(loopback.User);
        var userObj = (options.profileToUser || profileToUser)(provider, profile, options);
        if (!userObj.email && !options.emailOptional) {
-         process.nextTick(function() {
-           return cb('email is missing from the user profile');
-         });
+         return cb('email is missing from the user profile');
```

Comment:
@alexpls Please check this article to see the benefit of using `nextTick()`: http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony